bark/models/observer/observer_model.hpp

```diff
@@ -39,10 +39,12 @@ using world::WorldPtr;
 class ObserverModel : public commons::BaseType {
  public:
   explicit ObserverModel(bark::commons::ParamsPtr params)
-    : BaseType(params) {}
+    : BaseType(params),
+      observe_only_for_agents_() {}
```

Comment:
would rather name it `observed_agent_ids_`. Avoids confusion with using pointers

Reply:
Mhh, I think observe_only_for_agents_ is clearer since it describes what to is done with this information. observed_agent_ids_